apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsList/index.tsx

```diff
@@ -33,12 +36,13 @@ const CommitDetailsList: React.FC<CommitDetailsListProps> = ({
                 task={task}
               />
             );
-          } else if (t.inactiveTasks) {
-            // TODO DEVPROD-16174: Replace with Inactive Commits Button.
+          } else if (inactiveTasks && shouldCollapse) {
             return (
-              <span key={t.inactiveTasks[0].id} data-cy="collapsed-card">
-                {t.inactiveTasks.length} Collapsed
-              </span>
+              <InactiveCommitsButton
+                key={inactiveTasks[0].id}
```

Comment:
nit. could we `key` this by something else, perhaps the range 👀 I think the `key` might be duplicated due to also using it for the first card (if expanded).

e.g. `inactive-tasks-${inactiveTasks[0].id]}- ${inactiveTasks[inactiveTasks.length-1].id}` (or whatever else you can think of)